### ISSUE TITLE
Use builder for creating Job

### ISSUE THREAD

**Issue by wezm** [jonhoo/faktory-rs#9]
Currently setting properties on Job outside of kind and args requires manually building a Job or building a Job with new, then mutating its properties. I'd suggest using a builder to create jobs and make the fields of Job private. My suggested approach would probably be [derive_builder](https://crates.io/crates/derive_builder).

If this sounds reasonable I'm happy to make this change.

**Comment by jonhoo** [comment]
Yeah, that makes a lot of sense! I've been wanting to make that change in accordance with [`C-BUILDER`](https://rust-lang-nursery.github.io/api-guidelines/type-safety.html#c-builder) in the Rust API guidelines too. A PR using `derive_builder` seems reasonable to me!

**Comment by jonhoo** [comment]
@wezm still think you'd like to make this change? I'd be happy to review and merge a change like this!

**Comment by rustworthy** [comment]
Implemented with the help of `derive_builder` crate and released in `v0.13.0`
